### PR TITLE
🗃️ change the relation of `internship/course`

### PR DESCRIPTION
**Pull request by shkurata**
Use an `implicit many-to-many` relation between the `Course` and the `Internship` models

### CODE REVIEW

File: api/src/domain/course.ts [hunk 1]
```diff
@@ -6,15 +6,17 @@ import { prisma } from '~/prisma';
...
-      where: { id },
+    const courses = await prisma.course.findMany({
+      where: { internships: { some: { id } } },
     });
-    if (course == null) {
+    if (courses.length === 0) {
       throw new AppNotFoundError();
     }
-    return course;
+    return courses;
```
Comment (moeriki) [inline api/src/domain/course.ts:19]: - `id: string` is an internship id? Then this should be in the internship domain.
- I'm guessing it'd be way more performant to query a internship by id and then list its courses. It also makes more sense.
- An empty list of courses is not a `NotFoundError`. If an `id` doesn't reference an existing record, that would be a not found error.

File: api/prisma/seed.ts [hunk 2]
```diff
@@ -142,9 +144,17 @@ async function createInternshipInstance(
...
-    throw new Error('No users found');
+    // throw new Error('No users found');
+    users = [
+      await prisma.user.create({
+        data: {
+          name: 'Nikolay Neykov',
+          id: '68146a7d-0e8a-48f4-be83-e3f5388471bb',
+        },
+      }),
+    ];
```
Comment (moeriki) [inline api/prisma/seed.ts:157]: This doesn't look like it should've been committed?

**Review by moeriki — CHANGES_REQUESTED** [review]

File: api/src/domain/internship.ts [hunk 3]
```diff
@@ -35,3 +36,20 @@ export async function getInternshipById(id: string): Promise<InternshipModel> {
     handleServiceError(error);
   }
 }
+
+export async function getCoursesForInternship(
```
Comment (moeriki) [inline api/src/domain/internship.ts:40]: I'd prefer `getInternshipCourses`.

File: api/src/domain/internship.ts [hunk 4]
```diff
@@ -35,3 +36,20 @@ export async function getInternshipById(id: string): Promise<InternshipModel> {
...
+  internshipId: string,
```
Comment (moeriki) [inline api/src/domain/internship.ts:41]: `id: string` is fine. Within the scope of the internship service it's clearly an internship id.

File: api/src/domain/course.ts (hunk 1 above)
Comment (shkurata) [reply api/src/domain/course.ts]: It is updated.

File: api/src/domain/internship.ts [hunk 5]
```diff
@@ -35,3 +36,20 @@ export async function getInternshipById(id: string): Promise<InternshipModel> {
...
+): Promise<CourseModel[]> {
+  try {
+    const internship = await prisma.internship.findUnique({
+      where: { id: internshipId },
+      include: { courses: true },
```
Comment (moeriki) [inline api/src/domain/internship.ts:46]: This returns the full internship while we're only interested in the courses.

I think this should be…

```suggestion
      select: { courses: true },
```

**Review by moeriki — CHANGES_REQUESTED** [review]

File: api/src/domain/internship.ts (hunk 3 above)
Comment (shkurata) [reply api/src/domain/internship.ts]: Updated.

File: api/src/domain/internship.ts (hunk 4 above)
Comment (shkurata) [reply api/src/domain/internship.ts]: Updated.

**Review by moeriki — APPROVED** [review]

**Comment by moeriki** [comment]
Looks good 👍

**Comment by shkurata** [comment]
And what about the build fails?

**Pull request merged**